lib/gitgitgadget.ts

```diff
@@ -164,41 +169,29 @@ export class GitGitGadget {
         return true;
     }
 
-    // better parms would be to pass prinfo object
-    public async submit(pr: IPullRequestInfo, userInfo: IGitHubUser):
+    // Send emails only to the user
+    public async preview(pr: IPullRequestInfo, userInfo: IGitHubUser):
         Promise<string | undefined> {
 
-        // if known, add submitter to email chain
-        const ccSubmitter = userInfo.email ? `\nCc: ${
-            userInfo.name} <${userInfo.email}>` : "";
-        const description = `${pr.title}\n\n${pr.body}${ccSubmitter}`;
-
-        if (pr.baseOwner !== "gitgitgadget" || pr.baseRepo !== "git") {
-            throw new Error(`Unsupported repository: ${pr.pullRequestURL}`);
-        }
+        const send = async (mail: string): Promise<string> => {
+            const mbox = await parseMBox(mail);
+            mbox.cc = [];
+            mbox.to = userInfo.email;
+            return await sendMail(mbox, this.smtpOptions);
+        };
 
-        // get metadata in work repo
-        const metadata =
-            await this.notes.get<IPatchSeriesMetadata>(pr.pullRequestURL);
-        const previousTag = metadata && metadata.latestTag ?
-            `refs/tags/${metadata.latestTag}` : undefined;
-        // update work repo from base
-        await this.updateNotesAndPullRef(pr.number, previousTag);
+        return await this.genAndSend( pr, userInfo, {noUpdate: true}, send);
+    }
```

Comment:
In other words, I think that we will need to pass through e.g. a `coverMidSuffix` (that I would initialize to some known unique value for `prefix()`, e.g. `${Date.now()}`) and append it to the Message-ID [here](https://github.com/gitgitgadget/gitgitgadget/blob/bdcebdffd98610b6d9e295cf160b189d4e1d7746/lib/patch-series.ts#L637-L638).

Reply:
@dscho Git generates the emails with a unique value in the message-id.  What do you thing about retaining that in the new cover letter message id?  It is also part of the individual fix message-ids.  This would remove the need for a new option.  This would ensure multiple /preview comments get different message-id values.
ie
_Git generates_:
Message-Id: <cover.1570600244.git.user@example.com>
In-Reply-To: <pull.59.git.1570600233.user@example.com>
_gitgitgadget changes it to:_
Message-ID: <pull.59.v2.git.1570600244.user@example.com>
In-Reply-To: <pull.59.git.1570600233.user@example.com>

I understand the reply comment for /submit and /preview will not be as clean as it currently is but is that a problem?

Reply:
Instance extraction at ~627:
            const coverInst = coverMid.match(/cover(\.[0-9]+\.?)/);

Reply:
> What do you thing about retaining that in the new cover letter message id?

Hmm. I rather like that the Message-IDs talk about GitGitGadget (or at least about the originating PullRequest).

> _gitgitgadget changes it to:_
> Message-ID: [pull.59.v2.git.1570600244.user@example.com](mailto:pull.59.v2.git.1570600244.user@example.com)
> In-Reply-To: [pull.59.git.1570600233.user@example.com](mailto:pull.59.git.1570600233.user@example.com)

Actually, it does not. It changes it to `pull.59.git.gitgitgadget@gmail.com`, see e.g. https://public-inbox.org/git/pull.380.git.gitgitgadget@gmail.com/

How about changing the Message-ID to include the Unix epoch always (or even easier: the milliseconds since Jan 1st 1970, which is what `Date.now()` returns)? I.e. something like `pull.59.git.1570609920817.gitgitgadget@gmail.com`?

Reply:
> Hmm. I rather like that the Message-IDs talk about GitGitGadget (or at least about the originating PullRequest).

Agree.  I was just clumsily proposing to retain the milliseconds value provided by git.  That would keep it consistent with the rest of the patch emails.

> Actually, it does not. It changes it to `pull.59.git.gitgitgadget@gmail.com`, see e.g. https://public-inbox.org/git/pull.380.git.gitgitgadget@gmail.com/

Sorry, the example was with the proposed change and the unit tests require user@example.com.

> How about changing the Message-ID to include the Unix epoch always (or even easier: the milliseconds since Jan 1st 1970, which is what `Date.now()` returns)? I.e. something like `pull.59.git.1570609920817.gitgitgadget@gmail.com`?

Sounds like we agree on this.  I will use the timestamp from the git format-patch output.
Cover letter (user@example.com would be gitgitgadget on azure):
Message-Id: <pull.59.v2.git.1570638881.user@example.com
Patch email:
Message-Id: <f886cb837afaf10c720beae3ca88e740d93b3aea.1570638881.git.user@example.com

Reply:
> the example was with the proposed change

Sorry for being so thick! Yes, I agree with that change ;-)